subscription.go

```diff
@@ -308,8 +309,17 @@ func (s *Subscription) Unsubscribe() error {
 	return nil
 }
 
+// Close remove subscription from client's subs map
+func (s *Subscription) Close() {
+	s.status = SUBCLOSED
+	s.centrifuge.removeSubscription(s.channel)
+}
+
 // Subscribe allows to subscribe again after unsubscribing.
 func (s *Subscription) Subscribe() error {
+	if s.status == SUBCLOSED {
```

Comment:
Reading should be protected by `s.mu` lock - otherwise, we introduce data race here. Also status should be checked in every other Subscription methods like `History`, `Publish`, `Presence`.